lib/pharos/salt_api.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+require "net/http"
+
+module Pharos
+  # This class offers the integration between ruby and the Saltstack API.
+  module SaltApi
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    attr_accessor :token
+
+    # Perform request to endpoint, with method and provided data
+    def perform_request(endpoint:, method:, data: {})
+      self.class.perform_request endpoint: endpoint,
+                                 method:   method,
+                                 data:     data,
+                                 client:   self
+    end
+
+    # Defines class methods
+    module ClassMethods
+      # Performs a log in request and sets the @token instance variable on success at
+      # client object.
+      def login!(client:)
+        return client.token if client&.token
+        token = perform_request(endpoint: "/login",
+                                method:   "post",
+                                data:     auth_details)["X-Auth-Token"]
+        client&.token = token
+        token
+      end
+
+      # Returns the authentication details for the current environment.
+      def auth_details
+        {
+          username: ENV["PHAROS_SALT_USER"],
+          password: ENV["PHAROS_SALT_PASSWORD"],
+          eauth:    "pam"
+        }.freeze
+      end
+
+      # Returns salt hostname for the current environment.
+      def hostname
+        "#{ENV["PHAROS_SALT_HOST"]}:#{ENV["PHAROS_SALT_PORT"]}"
+      end
+
+      # Returns true if it is a tokenless request
```

Comment:
Maybe add this link to the comments?
https://docs.saltstack.com/en/latest/ref/netapi/all/salt.netapi.rest_cherrypy.html#run